### PR TITLE
Cypress v10 - Implement Cypress Component Tests

### PR DESCRIPTION
**Pull request by kevinold**

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -17,5 +17,5 @@
     "jsx": "react-jsx",
     "noFallthroughCasesInSwitch": true
   },
-  "include": ["src", "scripts", "backend", "src/__tests__"]
+  "include": ["src/**/*.spec.ct.{js,ts,jsx,tsx}", "scripts", "backend", "src/__tests__"]
```
Comment (cowboy) [inline tsconfig.json]: Use `src/**/*.cy.{js,ts,jsx,tsx}`

File: src/components/TransactionTitle.spec.ct.tsx [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import * as React from "react";
```
Comment (cowboy) [inline src/components/TransactionTitle.spec.ct.tsx]: Call this file `TransactionTitle.cy.tsx`

File: cypress/plugins/index.ts [hunk 3]
```diff
@@ -1,3 +1,5 @@
+/// <reference types="cypress" />
```
Comment (cowboy) [inline cypress/plugins/index.ts]: Ideally you'll just delete this file

**Comment by brian-mann** [comment]
@kevinold Ely and I are reviewing all of the in flight DX work and found this PR - we were both wondering what the status of this is. We see it's draft and based on the jira issue it appears you completed most if not all of the component tests. Do you mind giving us a quick update? Thanks

File: cypress/support/auth-provider-commands/auth0.ts [hunk 4]
```diff
@@ -3,7 +3,7 @@
 
 import * as jwt from "jsonwebtoken";
 
-Cypress.Commands.add("loginByAuth0Api", (username: string, password: string) => {
```
Comment (jordanpowell88) [inline cypress/support/auth-provider-commands/auth0.ts:6]: Why are you removing this explicit type?

**Review by jordanpowell88 — APPROVED** [review]

**Pull request merged**